tests/test_repo_synchronizer.py

```diff
@@ -215,9 +216,69 @@ def test_sync_process_duplicate_tags(
     assert tag in tag_log
 
 
+def test_sync_process_no_duplicate_tags_on_error(
+    git_source: Path,
+    hg_destination: Path,
+    tmp_path: Path,
+) -> None:
+    """Reprocessing a tag message after a failure should not lead to duplicate tags."""
+
+    tag_branch = "tags"
+    tag = "mytag"
+    tag_suffix = "some suffix"
+
+    repo = Repo(git_source)
+
+    git_commit_sha = repo.rev_parse("HEAD")
+
+    git_local_repo_path = tmp_path / "clones" / "myrepo"
+    syncrepos = RepoSynchronizer(git_local_repo_path, str(git_source))
+
+    operations: list[SyncBranchOperation | SyncTagOperation] = [
+        SyncTagOperation(
+            source_commit=git_commit_sha,
+            tag=tag,
+            tags_destination_branch=tag_branch,
+            tag_message_suffix=tag_suffix,
+        ),
+    ]
+
+    request_user = "request_user@example.com"
+
+    # Make the target repo unwriteable,, so the next syncs fail.
+    failhook = """
+[hooks]
+pretxnchangegroup.reject = /bin/false
+    """
```

Comment:
nit: this could be rewritten a bit better.
```suggestion
    failhook = "[hooks]\npretxnchangegroup.reject = /bin/false"
```

or another way without messing up indentation.